src/main/java/com/minecolonies/core/entity/other/cavalry/CavalryHorseEntity.java

```diff
@@ -0,0 +1,876 @@
+    }
+
+    /**
+     * Linearly interpolates between the current angle and the target angle, not exceeding the maximum turn angle per tick.
+     * 
+     * @param current        the current angle
+     * @param target         the target angle
+     * @param maxTurnPerTick the maximum angle to turn per tick
+     * @return the new angle after turning
+     */
+    private static float turnToward(float current, float target, float maxTurnPerTick)
+    {
+        float delta = net.minecraft.util.Mth.wrapDegrees(target - current);
+        if (delta > maxTurnPerTick) delta = maxTurnPerTick;
+        if (delta < -maxTurnPerTick) delta = -maxTurnPerTick;
+        return current + delta;
+    }
+
+    /** Allow breeding even when not tamed. Keep vanilla species rules. */
+    @Override
+    public boolean canMate(@Nonnull Animal other)
+    {
+        if (other == this) return false;
+        if (!(other instanceof AbstractHorse)) return false;
+        if (this.isBaby() || other.isBaby()) return false;
+
+        // vanilla enforces isTamed() here; we intentionally skip that
+        if (other instanceof Mule) return false;
+        if ((this instanceof Horse && (other instanceof Horse || other instanceof Donkey)))
+        {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * A slimmed version of the tick function that only runs on the server side.
+     * This function is responsible for updating the horse's slim state based on 
+     * whether it is being ridden by a guard or trainer. If the horse is being ridden, 
+     * it will be slimmed. Otherwise, it will not be slimmed unless it is unable to 
+     * widen safely, in which case it will be given a certain number of ticks to try 
+     * and move to a safe location.
+     */
+    protected void slimTick() 
+    {
+        if (!level().isClientSide) {
+
+            if (hasCavalryRider() || hasTrainer()) 
+            {
+                slimGraceTicks = SLIM_GRACE_MAX_TICKS;
+            }
+            else 
+            {
+                if (canWidenSafelyHere()) 
+                {
+                    slimGraceTicks = 0;
+                } 
+                else if (!(tryNudgeToWidenSpot() && canWidenSafelyHere())) 
+                {
+                    slimGraceTicks = Math.max(0, slimGraceTicks - 1);
+                } 
+                else 
+                {
+                    slimGraceTicks = 0;
+                }
+            }
+
+            boolean slimNow = shouldBeSlim() || (slimGraceTicks > 0 && !canWidenSafelyHere());
+            setSlim(slimNow);
+        }
+    }
+
+    /**
+     * Called every tick to update the horse. 
+     */
+    @Override
+    public void tick()
+    {
+        super.tick();
+        // logActiveGoals();
+
+        if (!level().isClientSide)
+        {
+            slimTick();
+
+            if (hasReservation())
+            {
+                if (reservationExpiration > RESERVATION_EXPIRATION_LIMIT) 
```

Comment:
these reservations I'd also have on the colony level. And we can "slow tick" them down there too just fine.